EasyReflectometry/calculators/refl1d/wrapper.py

```diff
@@ -155,61 +145,43 @@ def calculate(self, q_array: np.ndarray, model_name: str) -> np.ndarray:
-                structure |= model.Repeat(stack, repeat=i.repeat.value)
-
-        argmin = np.argmin(q_array)
-        argmax = np.argmax(q_array)
-        dq_vector = self._resolution_function(q_array)
+        sample = _build_sample(self.storage, model_name)
+        dq_array = self._resolution_function(q_array)
 
         if is_constant_resolution_function(self._resolution_function):
             # Get percentage of Q and change from sigma to FWHM
-            dq_vector = dq_vector * q_array / 100 / (2 * np.sqrt(2 * np.log(2)))
-
-        if MAGNETISM:
-            xs = []
-            for _ in range(4):
-                q = names.QProbe(
-                    Q=q_array,
-                    dQ=dq_vector,
-                    intensity=self.storage['model'][model_name]['scale'],
-                    background=self.storage['model'][model_name]['bkg'],
-                )
-                q.calc_Qo = np.linspace(
-                    q_array[argmin] - PADDING_RANGE * dq_vector[argmin],
-                    q_array[argmax] + PADDING_RANGE * dq_vector[argmax],
-                    UPSCALE_FACTOR * len(q_array),
-                )
-                xs.append(q)
-            probe = names.PolarizedQProbe(xs=xs, name='polarized')
-
-            R = names.Experiment(probe=probe, sample=structure).reflectivity()[1]
-        else:
-            q = names.QProbe(
-                Q=q_array,
-                dQ=dq_vector,
-                intensity=self.storage['model'][model_name]['scale'],
-                background=self.storage['model'][model_name]['bkg'],
+            dq_array = dq_array * q_array / 100 / (2 * np.sqrt(2 * np.log(2)))
+
+        if not MAGNETISM:
+            probe = _get_probe(
+                q_array=q_array,
+                dq_array=dq_array,
+                model_name=model_name,
+                storage=self.storage,
+                oversampling_factor=OVERSAMPLING_FACTOR,
             )
-            q.calc_Qo = np.linspace(
-                q_array[argmin] - PADDING_RANGE * dq_vector[argmin],
-                q_array[argmax] + PADDING_RANGE * dq_vector[argmax],
-                UPSCALE_FACTOR * len(q_array),
+            _, reflectivity = names.Experiment(probe=probe, sample=sample).reflectivity()
+        else:
+            polarized_probe = _get_polarized_probe(
+                q_array=q_array,
+                dq_array=dq_array,
+                model_name=model_name,
+                storage=self.storage,
+                oversampling_factor=OVERSAMPLING_FACTOR,
+                all_polarizations=ALL_POLARIZATIONS,
             )
-            R = names.Experiment(probe=q, sample=structure).reflectivity()[1]
+            polarized_reflectivity = names.Experiment(probe=polarized_probe, sample=sample).reflectivity()
+
+            if ALL_POLARIZATIONS:
+                raise NotImplementedError('Polarized reflectivity not yet implemented')
+                _, reflectivity_pp = polarized_reflectivity[0]
```

Comment:
The code underneath the exception is not reachable.
Please comment it out if needed for future.

Reply:
done